### ISSUE TITLE
help request: authz-keycloak plugin to support auth code flow (bearer_only: false) 

### ISSUE THREAD

**Issue by unreal-asr** [apache/apisix#9256]
### Description

From the plugin [documentation ](https://apisix.apache.org/docs/apisix/plugins/authz-keycloak/) it seems it is not possible to use the classic browser flow for NO-REST WEB applications.

With ApiSix [openid-connect](https://apisix.apache.org/docs/apisix/plugins/openid-connect/) plugins it is possible to set the bearer_only: false property but this plugin does not enforce the security policies (PEP) like authz-keycloak plugin so currently it seems not possible with ApiSix + plugins to protect a WEB application by applying the security policies without having to touch the source code of the WEB application to protect.

**Comment by unreal-altran** [comment]
Is support for bearer_only:false foreseen in the plugin roadmap?
Any News ?

**Comment by kayx23** [comment]
> classic browser flow

You mean authorization code flow?

**Comment by shreemaan-abhishek** [comment]
ping @unreal-altran

**Comment by kayx23** [comment]
> I would like to obtain the exact opposite behavior and manage a browser login guided by this specific plugin that also support the PEP policy.
> 
> It is not clear to me from the plugin documentation if and how it is possible to achieve this result.

I just tried to implement RBAC in keycloak and use it with `openid-connect` plugin but it didn't work out. I suspect it is not supported currently and found a related conversation here: https://github.com/zmartzone/lua-resty-openidc/issues/222

Here's how I set up keycloak:

* create realm, client, and user
* enable Authorization in the client
* create a realm role called `admin`
* create a resource and configure it with the URI I want to protect (same as the route URI) under Authorization
* create a role-based policy that requires the previously created realm role `admin`
* create a resource-based permission that maps the policy to the resource
* map the realm role `admin` to the user

So if user doesn't have the realm role `admin`, one could expect the access to be denied. In my testing, the request still went through even when the user doesn't have the realm role `admin`.

I'm just documenting what I have attempted here for future reference.

As an alternative, we could perhaps leverage scope-based access control with `required_scope`: https://github.com/apache/apisix/pull/10493
